src/tools/ceph_dedup/ceph_dedup_daemon.cc

```diff
@@ -662,16 +668,14 @@ int make_crawling_daemon(const po::variables_map &opts)
     << ")" 
     << dendl;
 
-  std::shared_lock l(glock);
+  state = new SampleDedupWorkerThread::SampleDedupGlobal(
+    chunk_dedup_threshold, sampling_ratio, report_period, fp_threshold);
```

Comment:
Apologies, I didn't look closely enough -- you don't even need a std::unique_ptr.  You can just declare it directly as
```
  SampleDedupWorkerThread::SampleDedupGlobal state(
    chunk_dedup_threshold, sampling_ratio, report_period, fp_threshold);
```
since it's a simple, concrete structure.